tls/ttls.c

```diff
@@ -2239,7 +2237,7 @@ ttls_recv(void *tls_data, unsigned char *buf, size_t len, unsigned int *read)
 	case TTLS_MSG_ALERT:
```

Comment:
I believe the block
```C
        delta = *read - parsed;                                                        
        len -= delta;                                                                  
        buf += delta;
        parsed = *read; 
```
should be under the `if (!(io->st_flags & TTLS_F_ST_HDRIV))` statement and before `if (io->msgtype == TTLS_MSG_HANDSHAKE...)`, so we can use `delta` instead of two `*read - parsed` expressions in another `if` - trivial change, but it makes the code simpler.

Reply:
Indeed. Moved the block into the if statement body.